webpages/settings/index.js

```diff
@@ -24,18 +24,25 @@ Vue.directive("click-outside", {
 });
 
 const ColorInput = Vue.extend({
-  props: ["value", "addon", "setting"],
+  props: ["value", "addon", "setting","no_alpha"],
   template: document.querySelector("template#picker-component").innerHTML,
   data() {
     return {
       isOpen: false,
       color: this.value,
       canCloseOutside: false,
+      formats: '',
     };
   },
   ready() {
-    self = this;
-    this.$els.pickr.addEventListener("change", (e) => {
+    self = this
```

Comment:
Is this creating a global variable?

Reply:
It is mutating window.self, can you use `const that = this` instead? (or directly use `this` if possible)